### PR TITLE
Auto Abort Changes

### PR DESCRIPTION
**Pull request by mrinal2601**

### CODE REVIEW

File: src/mappers.ts [hunk 1]
```diff
@@ -1,515 +1,585 @@
...
+) {
+  var data = {
+    testRunId: testRunName,
+    displayName: runDisplayName ? runDisplayName : getDefaultTestRunName(),
+    description: runDescription
+      ? runDescription
+      : "Started using GitHub Actions",
+    testId: testId,
+    secrets: secretsRun,
+    autoStopCriteria: autoStop,
```
Comment (radhikajain99) [inline src/mappers.ts:152]: nit: Test run will take the value from test so should we remove it from here?

File: src/mappers.ts [hunk 2]
```diff
@@ -1,515 +1,585 @@
...
+    errorRateTimeWindow: 60,
+  };
+
+  if (typeof autoStop == "string") {
+    data = {
+      isAutoAbortEnabled: false,
+      errorRate: 0,
+      errorRateTimeWindow: 0,
+    };
+  }
```
Comment (radhikajain99) [inline src/mappers.ts:576]: nit: Should we add validation for the autoStop string value to be "disable" ? And fail with error that auto stop is not valid if it is not correct?

File: src/mappers.ts [hunk 3]
```diff
@@ -1,515 +1,585 @@
...
+  if (autoStop) {
+    data = {
+      isAutoAbortEnabled: true,
+      errorRate: autoStop.errorRate,
+      errorRateTimeWindow: autoStop.errorRateTimeWindow,
+    };
```
Comment (radhikajain99) [inline src/mappers.ts:582]: nit: Do we need to add validations for error rate and errorRateTimeWindow?

**Review by radhikajain99 — DISMISSED** [review]

File: src/mappers.ts (hunk 1 above)
Comment (mrinal2601) [reply src/mappers.ts]: Done

File: src/mappers.ts (hunk 2 above)
Comment (mrinal2601) [reply src/mappers.ts]: updated

File: src/mappers.ts (hunk 3 above)
Comment (mrinal2601) [reply src/mappers.ts:583]: I think test will not be created if the value is not number.

**Review by radhikajain99 — APPROVED** [review]

**Pull request merged**